src/controllers/incident.controller.js

```diff
@@ -51,22 +51,13 @@ const incidentController = {
       return returnMessage(res, 500, 'Internal server error');
     }
   },
-  updateLocation: (req, res) => {
+  updateLocation: async (req, res) => {
     try {
       if (req.user.role === 'citizen') {
-        const index = incidents.findIndex(
-          (item) => item.incidentId.toString() === req.params.incidentId,
-        );
-        if (index > -1) {
-          if (incidents[index].status !== 'pending') {
-            return returnMessage(res, 404, {
-              message: 'You are not allowed to update this incident',
-            });
-          }
-
-          incidents[index].location = req.body.location;
+        const query = await db.updateIncident(parseInt(req.params.incidentId, 0), 'location', req.body.location, req.user.userid);
+        if (query.rowCount === 1) {
```

Comment:
Hi @shaazk learn how to use object destructuring, in order to avoid `req.params.incidentId or req.body.location` you can use object destructuring to achieve the same result with a clean syntax, check out this resource https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/